### DISCUSSION TITLE
Marking success or failled of a single or group of tasks effects other tasks under different execution dates

### DISCUSSION THREAD

**Discussion by rorymcstay** [apache/airflow#36925] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.3.3

### What happened?

When marking a task as success or fail from the UI (either browse task instances screen or from the grid) causes all tasks of the same task id to be also marked success.

Dags are scheduled using a timetable (`airflow.timetables.events.EventsTimetable`). Using postgres

No failed tasks initially. Mark Failed a single task instance is selected (Past/Future/Upstream/Downstream are NOT selected)
![image](https://github.com/apache/airflow/assets/33635190/89e85d2e-0ec0-4dd6-a8a7-29e7a7ed3c45)

UI Prompts to mark all failed
![image](https://github.com/apache/airflow/assets/33635190/26514c9e-729f-4b46-ad98-2e087ff30266)

All tasks marked failure
![image](https://github.com/apache/airflow/assets/33635190/531ab577-3339-454f-9323-3e0c053f1803)

### What you think should happen instead?

Only the tasks which were selected should be marked accordingly

### How to reproduce

1) Create a dag using the airflow EventsTimetable
2) Ensure a number of dag runs have been populated
3) Mark a task success or fail
4) Observe that the execution date is not respected

### Operating System

Ubuntu 22.04 LTS

### Versions of Apache Airflow Providers

$ pip freeze | grep apache-airflow-providers
apache-airflow-providers-common-sql==1.3.4
apache-airflow-providers-ftp==3.2.0
apache-airflow-providers-http==4.2.0
apache-airflow-providers-imap==3.1.0
apache-airflow-providers-sqlite==3.3.1

### Deployment

Other Docker-based deployment

### Deployment details

apache airflow 2.3.3 image is being used

### Anything else?

Problem occurs every time, even from the brow task instances screen, or marking an entire dag success/fail

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
I recall that issue has been fixed already - can't find it easily, as we had thousands of issues fixed since 2.3.3. Please update to latest version of Airflow (2.8.1) and try again the same scenario.

**Reply by rorymcstay** [reply]
Thanks, I did try to find a report on the issue but could not find anything. Thanks for the information, I will upgrade.

**Comment by potiuk** [comment]
Converting to discussion - in case more discussion is needed.
